scripts/src/javatests/org/oppia/android/scripts/label/AccessibilityLabelCheckTest.kt

```diff
@@ -0,0 +1,468 @@
+          package="org.oppia.android">
+          <activity
+              android:name=".app.home.HomeActivity" />
+      </manifest>
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main")
+    val tempFileRelativePath = "app/src/main/AndroidManifest.xml"
+    val manifestFile = tempFolder.newFile("testfiles/$tempFileRelativePath")
+    manifestFile.writeText(testContent)
+    val exemptionFile = File("${tempFolder.root}/$pathToProtoBinary")
+    val builder = AccessibilityLabelExemptions.newBuilder()
+    builder.addExemptedActivity("app/src/main/java/org/oppia/android/app/home/HomeActivity")
+    val exemptions = builder.build()
+    exemptions.writeTo(exemptionFile.outputStream())
+
+    main(
+      retrieveTestFilesDirectoryPath(),
+      "${tempFolder.root}/$pathToProtoBinary",
+      tempFileRelativePath
+    )
+
+    assertThat(outContent.toString().trim()).isEqualTo(
+      ACCESSIBILITY_LABEL_CHECK_PASSED_OUTPUT_INDICATOR
+    )
+  }
+
+  @Test
+  fun testAccessibilityLabel_addRedundantExemption_checkShouldFail() {
+    val testContent =
+      """
+      <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="org.oppia.android">
+          <activity
+              android:name=".app.home.HomeActivity"
+              android:label="@string/administrator_controls_title1" />
+      </manifest>
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main")
+    val tempFileRelativePath = "app/src/main/AndroidManifest.xml"
+    val manifestFile = tempFolder.newFile("testfiles/$tempFileRelativePath")
+    manifestFile.writeText(testContent)
+    val exemptionFile = File("${tempFolder.root}/$pathToProtoBinary")
+    val builder = AccessibilityLabelExemptions.newBuilder()
+    builder.addExemptedActivity("app/src/main/java/org/oppia/android/app/home/HomeActivity")
+    val exemptions = builder.build()
+    exemptions.writeTo(exemptionFile.outputStream())
+
+    val exception = assertThrows(Exception::class) {
+      main(
+        retrieveTestFilesDirectoryPath(),
+        "${tempFolder.root}/$pathToProtoBinary",
+        tempFileRelativePath
+      )
+    }
+
+    assertThat(exception).hasMessageThat().contains(
+      ACCESSIBILITY_LABEL_CHECK_FAILED_OUTPUT_INDICATOR
+    )
+    val failureMessage =
+      """
+      Redundant exemptions:
+      - app/src/main/java/org/oppia/android/app/home/HomeActivity
+      Please remove them from scripts/assets/accessibility_label_exemptions.textproto
+      """.trimIndent()
+    assertThat(outContent.toString().trim()).isEqualTo(failureMessage)
+  }
+
+  @Test
+  fun testAccessibilityLabel_addRedundantExemption_activityMissingLabel_allFailuresShouldLog() {
+    val testContent =
+      """
+      <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="org.oppia.android">
+          <activity
+              android:name=".app.home.SplashActivity" />
```

Comment:
Can you also add a test where the activity's reference is fully qualified rather than relative?

Reply:
This was already added, renamed it now to give a better context.